apps/src/applab/designProperties.jsx

```diff
@@ -48,6 +29,7 @@ var DesignProperties = module.exports = React.createClass({
     // TODO (brent) - right now if i create two elements with the same id, I
     // can still run into the same problem, where I click on the other element
     // and the table doesn't update
+    // TODO (brent) - it appears the wrong element sometimes gets deleted
     return (
       <div key={this.props.element.id}>
```

Comment:
we should probably consider enforcing unique element ids in design mode soon, since we are depending on them for React to function correctly. Sounds a little tricky because if I have an element with id "button" and I'm trying to set the id of another element by typing "button1" from scratch, there will be an intermediate state where react wants to set its id to "button", and if we reject that change then I will end up with "butto1" (no 'n'). It's possible we could revert the id to its original value when "Done" is clicked or the element is otherwise unselected.

Alternately we could use our own element numbering scheme which the user cannot edit.